bundles/org.openhab.binding.teleinfo/src/main/java/org/openhab/binding/teleinfo/internal/handler/TeleinfoAbstractElectricityMeterHandler.java

```diff
@@ -75,7 +77,10 @@ public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
             TeleinfoAbstractControllerHandler controllerHandler = (TeleinfoAbstractControllerHandler) bridge
                     .getHandler();
             if (controllerHandler != null) {
-                controllerHandler.addListener(this);
+                if (!isListeningBridge) {
+                    controllerHandler.addListener(this);
+                    isListeningBridge = true;
```

Comment:
You never set this back to false, but more importantly I don't see `controllerHandler.removeListener(this);` ever getting called.

Instead of using a `isListeningBridge` field to prevent duplicate adding of listeners. You should instead just change the `listeners` field in `TeleinfoAbstractControllerHandler` to a Set instead of a List. A `CopyOnWriteArraySet` would be a good set implementation to use instead of `Collections.synchronizedList(new ArrayList<>());`.